image_segmentation/pytorch/runtime/inference.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+from scipy import signal
+from tqdm import tqdm
+
+import torch
+import torch.nn.functional as F
+from torch.cuda.amp import autocast
```

Comment:
I think on the MLcommons inference we had to only include AMP iff cuda was available. Otherwise, it gave an error on CPU.

So another `.is_cuda_available()`

Reply:
`torch.cuda.*` are no-ops when cuda is not available. I will leave the code as it is